drivers/unix/file_access_unix.cpp

```diff
@@ -108,10 +108,8 @@ Error FileAccessUnix::open_internal(const String &p_path, int p_mode_flags) {
 			last_error = ERR_FILE_CANT_OPEN;
 			return last_error;
 		}
-		// Fix temporary file permissions (defaults to 0600 instead of 0666 & ~umask).
-		mode_t mask = umask(022);
-		umask(mask);
-		fchmod(fd, 0666 & ~mask);
+		umask(022);
```

Comment:
This change will set `umask` for the whole app, but the original code was intended to get the current mask instead (first call to get the mask, and second to restore it). 

`0022` is default `umask` on most Unix systems, but seems like web use `0777`. I guess it would be better to use `fchmod` with hardcoded `0666` and avoid touching `umask` at all, if it's causing issues.

Reply:
> This change will set umask for the whole app, but the original code was intended to get the current mask instead (first call to get the mask, and second to restore it).

Oh! I understand better your code, now.

Reply:
We could keep your code for Unix systems and hardcode 0666 for web with a macro.

Reply:
This part was there to mimic previous behavior (before code to create temp files were changed) as close as possible, but I do not think it's important for anything, so hardcoding 0666 everywhere should be fine as well.

Reply:
Done. I removed `umask()`.